CodeBeam.MudBlazor.Extensions/Components/RangeSlider/MudRangeSlider.razor.cs

```diff
@@ -0,0 +1,308 @@
+                    return;
+                }
+
+                if (Range && _upperValue != null && Convert.ToDecimal(d) >= Convert.ToDecimal(UpperValue))
+                {
+                    _userInvalidatedRange = true;
+                    return;
+                }
+
+                _value = d;
+                ValueChanged.InvokeAsync(value);
+            }
+        }
+
+        [Parameter]
+        [Category(CategoryTypes.Slider.Data)]
+        public T? UpperValue
+        {
+            get => Converter.Get(_upperValue);
+            set
+            {
+                var d = Converter.Set(value);
+                if (_upperValue == d)
+                {
+                    return;
+                }
+
+                if (Range && _value != null && Convert.ToDecimal(d) <= Convert.ToDecimal(Value))
+                {
+                    _userInvalidatedRange = true;
+                    return;
+                }
+
+                _upperValue = d;
+                UpperValueChanged.InvokeAsync(value);
+            }
+        }
+
+        /// <summary>
+        /// The color of the component. It supports the Primary, Secondary and Tertiary theme colors.
+        /// </summary>
+        [Parameter]
+        [Category(CategoryTypes.Slider.Appearance)]
+        public Color Color { get; set; } = Color.Primary;
+
+        protected string? Text
+        {
+            get => _value;
+            set
+            {
+                if (_value == value)
+                {
+                    return;
+                }
+                if (Range && Convert.ToDecimal(value) >= Convert.ToDecimal(UpperValue))
+                {
+                    _userInvalidatedRange = true;
+                    return;
+                }
+
+                _value = value;
+                ValueChanged.InvokeAsync(Value);
+            }
+        }
+
+        protected string? UpperText
+        {
+            get => _upperValue;
+            set
+            {
+                if (_upperValue == value)
+                {
+                    return;
+                }
+
+                if (Range && Convert.ToDecimal(value) <= Convert.ToDecimal(Value))
+                {
+                    _userInvalidatedRange = true;
+                    return;
+                }
+
+                _upperValue = value;
+                UpperValueChanged.InvokeAsync(UpperValue);
+            }
+        }
+
+        /// <summary>
+        /// If true, the dragging the slider will update the Value immediately.
+        /// If false, the Value is updated only on releasing the handle.
+        /// </summary>
+        [Parameter]
+        [Category(CategoryTypes.Slider.Behavior)]
+        public bool Immediate { get; set; } = true;
+
+        /// <summary>
+        /// If true, displays the slider vertical.
+        /// </summary>
+        [Parameter]
+        [Category(CategoryTypes.Slider.Appearance)]
+        public bool Vertical { get; set; } = false;
```

Comment:
It look like Vertical mode not working on visual, we can comment this paramater and can implement it later.